### ISSUE TITLE
Ignore scss maps within function-* rules

### ISSUE THREAD

**Issue by Nirazul** [stylelint/stylelint#983]
> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?

Bug (?): Sass Maps are throwing errors due to rules targeted for functions.

> Which rule, if any, is this issue related to?

`function-*` type rules

> What SCSS is needed to reproduce this issue?

e.g.

```scss
$media-expressions: (
    res2x: ('(min-resolution: 2dppx)'),
    res3x: ('(min-resolution: 3dppx)'),
);
```

> What stylelint configuration is needed to reproduce this issue?

e.g.

```json
{
  "rules": {
        "function-calc-no-unspaced-operator": true,
        "function-comma-newline-after": "always-multi-line",
        "function-comma-newline-before": "never-multi-line",
        "function-comma-space-after": "always",
        "function-comma-space-before": "never",
        "function-parentheses-newline-inside": "never-multi-line",
        "function-parentheses-space-inside": "never-single-line",
        "function-url-quotes": "single",
        "function-whitespace-after": "always",
  }
}
```

> Which version of stylelint are you using?

e.g. `5.2.1`

> How are you running stylelint: CLI, PostCSS plugin, Node API?

PostCSS with stylelint plugin.

> Does your issue relate to non-standard syntax (e.g. SCSS, nesting, etc.)?

Yes, it's related to SCSS maps.

> What did you expect to happen?

No warnings to be flagged.

> What actually happened (e.g. what warnings or errors you are getting)?

The following warnings were flagged:

```shell
_test.scss
1:22	⚠  Unexpected whitespace after "(" in a multi-line function (function-parentheses-newline-inside) [stylelint]
2:39	⚠  Expected single space after "," (function-comma-space-after) [stylelint]
2:40	⚠  Unexpected whitespace before ")" in a multi-line function (function-parentheses-newline-inside) [stylelint]
```

**Comment by davidtheclark** [comment]
It would be helpful if you could narrow down your list and identify which particular rules are causing you problems. Is it just `function-parentheses-newline-inside` and `function-comma-space-after`? We have fixed issues with Sass maps before, so I know that not *every* function rule will trip up on them :)

Also, those two rules seem like they're having unrelated problems, so maybe they deserve separate issues.

Finally: of course a PR solving the problem is the best way to get this done, but if you're unable to do that a PR including failing tests is extremely useful and will ensure this gets taken care of very quickly.

**Comment by Nirazul** [comment]
Ok, I'll try to narrow it down to distinct rules and maybe I'll find enough time for a PR that includes a few failing tests. Fixing the issue myself is beyond my knowledge and time horizon, i fear :(
But I'll try to help as much as I can!

**Comment by jeddy3** [comment]
@Nirazul Here's a pointer or two to help you out :) We wrote [a utility function](https://github.com/stylelint/stylelint/blob/master/src/utils/cssDeclarationIsMap.js) that determines if a declaration is a scss map. You can use this utility to return early in each of the function rules. There's an example of how to do this in [declaration-colon-newline-after](https://github.com/stylelint/stylelint/blob/master/src/rules/declaration-colon-newline-after/index.js#L30-L32). It looks like we've only used this utility in the aforementioned rule and [`declaration-colon-space-after`](https://github.com/stylelint/stylelint/blob/master/src/rules/declaration-colon-space-after/index.js), and so you're right that it might be most if not all of the `function-*` rules that need updating.

**Comment by Nirazul** [comment]
@jeddy3 Thanks for the great pointers. I was able to add all necessary tests to the `function-*` rules and was even able fix the affected rules!
